--- src/src/m1qn3.f	2009-10-20 03:39:35
+++ m1qn3.f	2024-05-06 11:12:30
@@ -802,7 +802,7 @@
      &        "  iter  simul  stepsize            f                |g|",
      &        "       |g|/|g0|"
           write(io,
-     &        '(1x,i5,2x,i5,2x,1pd8.2,2x,d21.14,2x,d11.5,2x,d10.4)')
+     &        '(1x,i5,2x,i5,2x,1pd9.2,2x,d21.14,2x,d12.5,2x,d11.4)')
      &        niter, isim, t, f, gnorm, eps1
       endif
       if (impres.ge.5) write (io,940) eps1
@@ -1355,7 +1355,8 @@
       if (tg.eq.0.d0) go to 940
       fn=fg
       do 930 i=1,n
-  930 xn(i)=xn(i)+tg*d(i)
+        xn(i)=xn(i)+tg*d(i)
+  930 end do
   940 if (imp.le.3) go to 999
       write (io,1001)
       write (io,1005) tg,fg,fpg
@@ -1366,7 +1367,8 @@
 c               recopiage de x et boucle
 c
   950 do 960 i=1,n
-  960 x(i)=xn(i)+t*d(i)
+        x(i)=xn(i)+t*d(i)
+  960 end do
       go to 100
 c     --- linesearch finished, no skip at next entry in mlis3
   999 if (reverse.ne.0) reentry = 0
