--- src/src/m1qn3.f	2009-10-20 03:39:35
+++ m1qn3.f	2024-05-29 11:49:22
@@ -124,7 +124,7 @@
 c
 c --- function
 c
-      double precision ddot,dnrmi
+      double precision ddot_m1qn3,dnrmi
 c
 c --- stop if reverse < 0 (m1qn3 should not be called with reverse < 0)
 c
@@ -373,7 +373,7 @@
      &        /5x,"number of simulations: ",i13
      &        /5x,"realized relative precision on g: ",1pd9.2)
       if (normtype.eq.'two') then
-          gnorm = sqrt(ddot(n,g,1,g,1))
+          gnorm = sqrt(ddot_m1qn3(n,g,1,g,1))
       elseif (normtype.eq.'sup') then
           gnorm = dnrmi(n,g)
       elseif (normtype.eq.'dfn') then
@@ -432,7 +432,7 @@
 c
 c         function
 c
-      double precision ddot,dnrmi
+      double precision ddot_m1qn3,dnrmi
 c
 c --- possible jumps
 c     9998: call of the simulator in m1qn3a with indic = 1
@@ -462,7 +462,7 @@
       call prosca (n,g,g,ps,izs,rzs,dzs)
       gnorm = dsqrt(ps)
       if (normtype.eq.'two') then
-          gnorms = sqrt(ddot(n,g,1,g,1))
+          gnorms = sqrt(ddot_m1qn3(n,g,1,g,1))
       elseif (normtype.eq.'sup') then
           gnorms = dnrmi(n,g)
       elseif (normtype.eq.'dfn') then
@@ -789,7 +789,7 @@
 c
       call prosca(n,g,g,ps,izs,rzs,dzs)
       if (normtype.eq.'two') then
-          gnorm = sqrt(ddot(n,g,1,g,1))
+          gnorm = sqrt(ddot_m1qn3(n,g,1,g,1))
       elseif (normtype.eq.'sup') then
           gnorm = dnrmi(n,g)
       elseif (normtype.eq.'dfn') then
@@ -802,7 +802,7 @@
      &        "  iter  simul  stepsize            f                |g|",
      &        "       |g|/|g0|"
           write(io,
-     &        '(1x,i5,2x,i5,2x,1pd8.2,2x,d21.14,2x,d11.5,2x,d10.4)')
+     &        '(1x,i5,2x,i5,2x,1pd9.2,2x,d21.14,2x,d12.5,2x,d11.4)')
      &        niter, isim, t, f, gnorm, eps1
       endif
       if (impres.ge.5) write (io,940) eps1
@@ -1355,7 +1355,8 @@
       if (tg.eq.0.d0) go to 940
       fn=fg
       do 930 i=1,n
-  930 xn(i)=xn(i)+tg*d(i)
+        xn(i)=xn(i)+tg*d(i)
+  930 end do
   940 if (imp.le.3) go to 999
       write (io,1001)
       write (io,1005) tg,fg,fpg
@@ -1366,7 +1367,8 @@
 c               recopiage de x et boucle
 c
   950 do 960 i=1,n
-  960 x(i)=xn(i)+t*d(i)
+        x(i)=xn(i)+t*d(i)
+  960 end do
       go to 100
 c     --- linesearch finished, no skip at next entry in mlis3
   999 if (reverse.ne.0) reentry = 0
