--- Makefile.am	2011-04-14 08:13:48.000000000 -0700
+++ Makefile.am_tmp	2012-11-12 14:54:02.000000000 -0800
@@ -24,10 +24,22 @@
 noinst_HEADERS = templates_on.h templates_off.h build.h
 
 MINGW32_HOST = @MINGW32_HOST@
-if MINGW32_HOST
+
+# Origional 'Makefile.am' sets 'libgslcblas.la' as a requirement for
+#'libgsl.ls' only if the host system is detected to be MingW32. This
+# is unfortunate as 'libgsl.la' has undefined 'cblas' symbols the
+# result is linking errors at run time. This patch sets the 'cblas'
+# library as a requirement for linking 'libgsl'.
+#
+# Origional script:
+#
+# if MINGW32_HOST
+# libgsl_la_LIBADD += cblas/libgslcblas.la
+# libgsl_la_LDFLAGS += -no-undefined
+# endif
+
 libgsl_la_LIBADD += cblas/libgslcblas.la
 libgsl_la_LDFLAGS += -no-undefined
-endif
 
 m4datadir = $(datadir)/aclocal
 m4data_DATA = gsl.m4
